otel.go

```diff
@@ -12,41 +13,50 @@ import (
 	"go.opentelemetry.io/otel/sdk/resource"
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
 	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/credentials"
+	otelTrace "go.opentelemetry.io/otel/trace"
+	"sync"
 )
 
 const (
-	packageName = "github.com/evergreen-ci/logkeeper"
+	packageName = "github.com/evergreen-ci/logkeeper/%s"
+	defaultName = "noop_tracer"
 )
 
-func (lk *logkeeper) initOtel(ctx context.Context) error {
-	if lk.opts.TraceCollectorEndpoint == "" {
-		// defaults to NoopTracerProvider
-		lk.tracer = otel.GetTracerProvider().Tracer(packageName)
-		return nil
-	}
+var (
+	closers       []closerOp
+	creator       sync.Once
+	closer        sync.Once
+	useCustomName bool
+)
 
+func LoadTraceProvider(ctx context.Context, useInsecure bool, collectorEndpoint string, sampleRatio float64) {
+	if collectorEndpoint == "" {
+		return
+	}
 	r, err := serviceResource(ctx)
 	if err != nil {
-		return errors.Wrap(err, "making host resource")
+		grip.EmergencyFatal(errors.Wrap(err, "making host resource"))
 	}
-
-	lk.otelGrpcConn, err = grpc.DialContext(ctx,
-		lk.opts.TraceCollectorEndpoint,
-		grpc.WithTransportCredentials(credentials.NewTLS(nil)),
-	)
-	if err != nil {
-		return errors.Wrapf(err, "opening gRPC connection to '%s'", lk.opts.TraceCollectorEndpoint)
+	var opts []otlptracegrpc.Option
+	if useInsecure {
```

Comment:
When would we want to not use TLS?

Reply:
I figured it would be useful if we ever want to collect traces from devs' machines or other test conditions